src/lib/swapper/swappers/utils/helpers/helpers.ts

```diff
@@ -152,30 +105,56 @@ export const getFeesFromFeeData = async ({
     return { gasLimit, maxFeePerGas, maxPriorityFeePerGas }
   if (gasPrice) return { gasLimit, gasPrice }
 
-  throw new SwapError('[getFeesFromFeeData]', {
+  throw new SwapError('[getFeesFromContractData]', {
     cause: 'legacy gas or eip1559 gas required',
     code: SwapErrorType.SIGN_AND_BROADCAST_FAILED,
   })
 }
 
-export const buildAndBroadcast = async ({
+export const createBuildCustomTxInput = async ({
   accountNumber,
   adapter,
-  data,
-  feeData,
-  to,
-  value,
+  erc20ContractAddress,
+  erc20ContractData,
+  sendAmountCryptoBaseUnit,
+  wallet,
+}: {
+  accountNumber: number
+  adapter: EvmChainAdapter
+  erc20ContractAddress: string
+  erc20ContractData: string
+  sendAmountCryptoBaseUnit: string
+  wallet: HDWallet
+}): Promise<evm.BuildCustomTxInput> => {
+  const gasFees = await getFeesFromContractData({
+    accountNumber,
+    adapter,
+    erc20ContractAddress,
+    erc20ContractData,
+    wallet,
+  })
+
+  return {
+    accountNumber,
+    data: erc20ContractData,
+    to: erc20ContractAddress,
+    value: sendAmountCryptoBaseUnit,
+    wallet,
+    ...gasFees,
+  }
+}
+
+export const buildAndBroadcast = async ({
```

Comment:
this is now very similar to the `defi/helpers/utils.ts` so we can unify all of the gas/tx building logic as a next improvement